doc/sonic-build-system/web_file_server_population_script.md

```diff
@@ -0,0 +1,127 @@
+8. [Q&A](#paragraph7)
+
+
+
+## Overview
+The SONiC reproducible build framework provides a way to lock the versions of external packages. It will lock the versions for all the docker images including the slave docker images, host images, and docker slave containers which are used to build the SONiC targets, such as deb files, python wheels, etc.
+SONiC reproducible build allows to identify specific version of a web package retrieved by wget/curl and download it from an external file storage. 
+The web site may be not stable, and the web packages may be replaced to another one unexpectedly, so we do not want to depend on the web packages at a remote web site. 
+Before a web package is used in the SONiC repository, the package should be uploaded to a trusted file storage. Currently SONiC offers to do that manually or automatically via Jenkins pipeline.
+The file Server population script is a complementary utility for “SONiC reproducible build” and suppose to ease the process of downloading the web packages from an external file storage and uploading them to trusted file storage.
+The script will be triggered manually or part of the build flow (part of “make freeze”) whenever the developer intend to populate the web server which packages associated to specific tag. 
+
+### In scope:
+- Download web packages from external file storage
+- Rename and upload web packages to trusted file storage
+- Script output
+
+## Download web packages from external file storage
+The script will identify all relevant web packages, collect their hash value and url and then download them from the remote site to a local cache.
+The script search over the source tree path recursively and look for ‘version-web’ files.
+Those file are being parsed in order to retrieve the path to the package and its hash value. The script is expecting for such format:
+[package url]==[hash value]
+
+```sh
+For example:
+https://storage.googleapis.com/golang/go1.14.2.linux-amd64.tar.gz==ebef065e4d35572af5b03e2be957a9c6c5063b38
+```
+
+### Work Items:
+- Identify relevant versions-web files
+- Parse versions-web files and identify relevant web packages
+- Download web packages from their external file storage
+
+
+## Upload web packages to trusted file storage
+
+Although there is no version for a package downloaded by wget/curl, the hash value of the package file can be used as the version value. 
+The script would rename the Web package and then upload it from cache to a trusted file server. The file name format in the file storage is as below:
+[original file name]-[hash value]
+```sh
+For an example: 
+go1.14.2.linux-amd64.tar.gz-ebef065e4d35572af5b03e2be957a9c6c5063b38
+```
+For the same web package, it supports to have multiple versions of the file in the file storage
+for example:
+```sh
+For an example: 
+go1.14.2.linux-amd64.tar.gz-ebef065e4d35572af5b03e2be957a9c6c5063b38
+go1.14.2.linux-amd64.tar.gz-b73d6366c0bce441d20138c150e2ddcdf406e654 (fake hash value)

```

Comment:
I believe it shouldn't be called as fake but a different hash value

Reply:
I will remove the comment